### PR TITLE
Force term elaboration

### PR DESCRIPTION
**Pull request by affeldt-aist**
##### Motivation for this change

Cyril introduced this notation in MathComp-Analysis based on an idea by Enrico
and it has since be scheduled for backport (see https://github.com/math-comp/analysis/issues/5).
We have been using it for a while in `posnum.v`, `nngnum.v`, and now `signed.v`.
This is just a copy-paste, together with the documentation written by Pierre.

@CohenCyril @gares @proux01 

##### Things done/to do

- [x] added corresponding entries in `CHANGELOG_UNRELEASED.md` (do not edit former entries)
- [x] added corresponding documentation in the headers

##### Automatic note to reviewers

Read [this Checklist](https://github.com/math-comp/math-comp/wiki/Checklist-for-following,-reviewing-and-playing-with-a-PR#checklist-for-reviewing-a-pr) and make sure there is a milestone.

### CODE REVIEW

File: mathcomp/ssreflect/ssreflect.v
Comment (gares) [inline mathcomp/ssreflect/ssreflect.v:19]:
```suggestion
```

**Review by gares — APPROVED** [review]

**Comment by affeldt-aist** [comment]
>     1. do we want to keep `!!` as the notation for this. It was the notation in mca until now, but I'm not fond of it.
>         Are there better syntax suggestions for this?

No idea. I got accustomed to it.

>     2. It has to be backported to coq/coq

So we should issue after merge to remind the release managers not to forget to backport the new version of `ssreflect.v` I guess.

**Comment by CohenCyril** [comment]
We should call for alternative proposals for this syntax and vote for the best. Also we should see if there are alternative plans that are less hackish. CC @gares

**Comment by affeldt-aist** [comment]
Since we didn't have a better proposition for the syntax, should we merge it as it is?

**Comment by gares** [comment]
`[infer t]` or `[fill t]` or `[elaborate t]`

File: CHANGELOG_UNRELEASED.md
Comment (gares) [inline CHANGELOG_UNRELEASED.md:113]:
```suggestion
  + notation `[elaborate t]` forcing the elaboration of `t` using Coq's `refine` tactic. This notation can be used in tandem with `have` to force type class resolution when an explicit proof term `t` is provided, instead of having type class instances be quantified implicitly.
```

**Comment by gares** [comment]
@CohenCyril please review the suggestion and merge

File: CHANGELOG_UNRELEASED.md
Comment (gares) [inline CHANGELOG_UNRELEASED.md:114]:
```suggestion
  + notation `[elaborate t]` forcing the elaboration of `t` using Coq's `refine` tactic.
    This notation can be used in tandem with `have` to force type class resolution when an
    explicit proof term `t` is provided (otherwise type class instances are quantified implicitly by `have`).

```

File: CHANGELOG_UNRELEASED.md
Comment (gares) [inline CHANGELOG_UNRELEASED.md:115]:
```suggestion
  + notation `[elaborate t]` forcing the elaboration of `t` using Coq's `refine` tactic.
    This notation can be used in tandem with `have` to force type class resolution
    when an explicit proof term `t` is provided (otherwise type class instances are
    quantified implicitly by `have`).
```

**Pull request merged**
